app/js/actions/jobs.js

```diff
@@ -1,13 +1,94 @@
-export const createJob = (job) => ({
-    type: 'CREATE_JOB',
+import actions from '../actions';
+import { makeB64Digest } from '../util/auth';
+
+// global variable to set and clear intervals for scanning for jobs.
+let jobInterval;
+
+export const newActiveJob = (workflow, job) => ({
+    type: 'NEW_ACTIVE_JOB',
+    workflow,
     job
 });
 
-export const startJob = (job) => ({
-    type: 'START_JOB',
+export const jobCompleted = (job) => ({
+    type: 'JOB_COMPLETED',
     job
 });
 
+export const pollJobStatus = () => {
+    return (dispatch, getState) => {
+        const { connection: { uri, availableApis } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}jobs`;
+        fetch(url, {
+            method: 'GET'
+        })
+        .then((response) => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            if (json.completed.length !== 0) {
+                json.completed.map(job => {
+                    if (Object.keys(job).length !== 0) {
+                        dispatch(actions.jobCompleted(job));
+                        dispatch(actions.refreshArtifacts());
```

Comment:
Also I am getting duplicate artifacts in my store when I run more than one job. This seems to happen because `refreshArtifacts` assumes every artifact returned from the server is a `newArtifact`, not entirely sure why, the logic seems fine in that action...